dev-test/gql-tag-operations-urql/gql/fragment-masking.ts

```diff
@@ -46,3 +48,19 @@ export function makeFragmentData<F extends DocumentTypeDecoration<any, any>, FT
 ): FragmentType<F> {
   return data as FragmentType<F>;
 }
+export function isFragmentReady<TQuery, TFrag>(
+  queryNode: DocumentTypeDecoration<TQuery, any>,
+  fragmentNode: TypedDocumentNode<TFrag>,
+  data: FragmentType<TypedDocumentNode<Incremental<TFrag>, any>> | null | undefined
+): data is FragmentType<typeof fragmentNode> {
+  const deferredFields = (queryNode as { __meta__?: { deferredFields: Record<string, (keyof TFrag)[]> } }).__meta__
+    ?.deferredFields;
+
+  if (!deferredFields) return true;
+
+  const fragDef = fragmentNode.definitions[0] as FragmentDefinitionNode | undefined;
+  const fragName = fragDef?.name?.value;
+
+  const fields = fragName ? deferredFields[fragName] : [];
+  return fields.length > 0 && fields.every(field => data && field in data);
```

Comment:
While testing the release candidate which includes this PR, I realised that if an user has `noUncheckedIndexedAccess` turned on in their `tsconfig.son`, the `fields` variable might be undefined, hence doing `fields.length` or `fields.every` fails and TS complains about it. Not sure how to fix this. Should I open an issue? Also apologies if this is not the place to comment about this.

@beerose

Reply:
Thanks for the feedback! You can open an issue and I'll improve this function to work with noUncheckedIndexedAccess.

Reply:
Here: https://github.com/dotansimha/graphql-code-generator/issues/9383. Thanks @beerose! I am happy to help testing this if needed. Cheers!

Reply:
> Here's a fix: #9385

Thanks!!